app/script/conversation/ConversationEphemeralHandler.js

```diff
@@ -30,48 +30,68 @@ z.conversation.ConversationEphemeralHandler = class ConversationEphemeralHandler
   }
 
   constructor(conversationService, onMessageTimeout) {
+    this._updateTimedMessages = this._updateTimedMessages.bind(this);
+
     this.conversationService = conversationService;
     this.onMessageTimeout = onMessageTimeout;
     this.logger = new z.util.Logger('z.conversation.ConversationEphemeralHandler', z.config.LOGGER.OPTIONS);
 
-    this.timedMessages = [];
-    window.setInterval(this._updateTimedMessages.bind(this), ConversationEphemeralHandler.CONFIG.INTERVAL_TIME);
+    this.timedMessages = ko.observableArray([]);
+
+    let updateIntervalId = null;
+    this.timedMessages.subscribe(messageEntities => {
+      const shouldClearInterval = !messageEntities.length && updateIntervalId;
+      if (shouldClearInterval) {
+        window.clearInterval(updateIntervalId);
+        updateIntervalId = null;
+        return this.logger.info('Cleared ephemeral message check interval');
+      }
+
+      const shouldSetInterval = messageEntities.length && !updateIntervalId;
+      if (shouldSetInterval) {
+        updateIntervalId = window.setInterval(() => {
+          this._updateTimedMessages();
+        }, ConversationEphemeralHandler.CONFIG.INTERVAL_TIME);
+        this.logger.info('Started ephemeral message check interval');
+      }
+    });
   }
 
   _updateTimedMessages() {
-    const now = Date.now();
-    for (let index = this.timedMessages.length - 1; index >= 0; index--) {
-      const messageEntity = this.timedMessages[index];
+    const currentTimestamp = Date.now();
+    const removeTimedMessage = (messageEntity, index) => {
+      amplify.publish(z.event.WebApp.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, messageEntity);
+      this._timeoutEphemeralMessage(messageEntity);
+      this.timedMessages().splice(index, 1);
```

Comment:
This might be a problem. You are mutating the array inside the observable directly, so the subscribtion won't trigger.